exercises/practice/two-bucket/src/test/java/TwoBucketTest.java

```diff
@@ -2,77 +2,110 @@
+        assertThat(bucketResult.getOtherBucket()).isEqualTo(3);
 
     }
 
     @Disabled("Remove to run test")
     @Test
     public void testBucketOneSizeSevenBucketTwoSizeElevenStartWithOne() {
 
-        TwoBucket twoBucket = new TwoBucket(7, 11, 2, "one");
+        Result bucketResult = new TwoBucket(7, 11, 2, "one").getResult();
 
-        assertThat(twoBucket.getTotalMoves()).isEqualTo(14);
-        assertThat(twoBucket.getFinalBucket()).isEqualTo("one");
-        assertThat(twoBucket.getOtherBucket()).isEqualTo(11);
+        assertThat(bucketResult.getTotalMoves()).isEqualTo(14);
+        assertThat(bucketResult.getFinalBucket()).isEqualTo("one");
+        assertThat(bucketResult.getOtherBucket()).isEqualTo(11);
 
     }
 
     @Disabled("Remove to run test")
     @Test
     public void testBucketOneSizeSevenBucketTwoSizeElevenStartWithTwo() {
 
-        TwoBucket twoBucket = new TwoBucket(7, 11, 2, "two");
+        Result bucketResult = new TwoBucket(7, 11, 2, "two").getResult();
 
-        assertThat(twoBucket.getTotalMoves()).isEqualTo(18);
-        assertThat(twoBucket.getFinalBucket()).isEqualTo("two");
-        assertThat(twoBucket.getOtherBucket()).isEqualTo(7);
+        assertThat(bucketResult.getTotalMoves()).isEqualTo(18);
+        assertThat(bucketResult.getFinalBucket()).isEqualTo("two");
+        assertThat(bucketResult.getOtherBucket()).isEqualTo(7);
 
     }
 
     @Disabled("Remove to run test")
     @Test
     public void testBucketOneSizeOneBucketTwoSizeThreeStartWithTwo() {
 
-        TwoBucket twoBucket = new TwoBucket(1, 3, 3, "two");
+        Result bucketResult = new TwoBucket(1, 3, 3, "two").getResult();
 
-        assertThat(twoBucket.getTotalMoves()).isEqualTo(1);
-        assertThat(twoBucket.getFinalBucket()).isEqualTo("two");
-        assertThat(twoBucket.getOtherBucket()).isEqualTo(0);
+        assertThat(bucketResult.getTotalMoves()).isEqualTo(1);
+        assertThat(bucketResult.getFinalBucket()).isEqualTo("two");
+        assertThat(bucketResult.getOtherBucket()).isEqualTo(0);
 
     }
 
     @Disabled("Remove to run test")
     @Test
     public void testBucketOneSizeTwoBucketTwoSizeThreeStartWithOne() {
 
-        TwoBucket twoBucket = new TwoBucket(2, 3, 3, "one");
+        Result bucketResult = new TwoBucket(2, 3, 3, "one").getResult();
 
-        assertThat(twoBucket.getTotalMoves()).isEqualTo(2);
-        assertThat(twoBucket.getFinalBucket()).isEqualTo("two");
-        assertThat(twoBucket.getOtherBucket()).isEqualTo(2);
+        assertThat(bucketResult.getTotalMoves()).isEqualTo(2);
+        assertThat(bucketResult.getFinalBucket()).isEqualTo("two");
+        assertThat(bucketResult.getOtherBucket()).isEqualTo(2);
+
+    }
+
+    @Disabled("Remove to run test")
+    @Test
+    public void testReachingGoalIsImpossible() {
+        
+        assertThatExceptionOfType(UnreachableGoalException.class)
+                .isThrownBy(() -> new TwoBucket(6, 15, 5, "one"))
```

Comment:
I think we want to call the `getResult()` in the assertion because I think it is also valid to do calculations in the `getResults` method instead of the constructor.

```suggestion
                .isThrownBy(() -> new TwoBucket(6, 15, 5, "one").getResult())
```

Reply:
I’ve updated the assertion to call `getResult()` as suggested. Let me know if there’s anything else you’d like adjusted!